api/nodes/views.py

```diff
@@ -106,7 +115,10 @@ def perform_destroy(self, instance):
         user = self.request.user
         auth = Auth(user)
         node = self.get_object()
-        node.remove_node(auth=auth)
+        try:
+            node.remove_node(auth=auth)
+        except NodeStateError as err:
```

Comment:
I purposefully did not handle the OSF `PermissionsError` because it should never be raised at this point due to the `ContributorOrPublic` permissions class.

Since we don't expect the `PermissionsError`, a loud error (i.e. a 500) should be raised if it does happen. This will help us identify that there's a bug in our permissions handling.

I am open to changing it though if you think otherwise.